pydicom/tests/test_valuerep.py

```diff
@@ -46,38 +48,155 @@ def test_pickling(self):
         assert x.original_string == x2.original_string
         assert str(x) == str(x2)
 
+    def test_str(self):
+        """Test str(TM)."""
+        x = pydicom.valuerep.TM("212223")
+        assert "212223" == str(x)
+        del x.original_string
+        assert not hasattr(x, 'original_string')
+        assert "21:22:23" == str(x)
+
+    def test_new_empty_str(self):
+        """Test converting an empty string."""
+        x = pydicom.valuerep.TM('')
+        assert x is None
+
+    def test_new_str_conversion(self):
+        """Test converting strings to times."""
+        x = pydicom.valuerep.TM('00')
+        assert "00" == str(x)
+        assert time(0, 0, 0) == x
+        x = pydicom.valuerep.TM('23')
+        assert "23" == str(x)
+        assert time(23, 0, 0) == x
+        msg = r"Unable to convert non-conformant value '24' to 'TM' object"
+        with pytest.raises(ValueError, match=msg):
+            pydicom.valuerep.TM('24')
+
+        x = pydicom.valuerep.TM('0000')
+        assert "0000" == str(x)
+        assert time(0, 0, 0) == x
+        x = pydicom.valuerep.TM('2359')
+        assert "2359" == str(x)
+        assert time(23, 59, 0) == x
+        msg = r"Unable to convert non-conformant value '2360' to 'TM' object"
+        with pytest.raises(ValueError, match=msg):
+            pydicom.valuerep.TM('2360')
+
+        x = pydicom.valuerep.TM('000000')
+        assert "000000" == str(x)
+        assert time(0, 0, 0) == x
+        # Valid DICOM TM seconds range is 0..60, but time is 0..59
+        msg = (
+            r"'datetime.time' doesn't allow a value of '60' for the "
+            r"seconds component, changing to '59'"
+        )
+        with pytest.warns(UserWarning, match=msg):
+            x = pydicom.valuerep.TM('235960')
+        assert "235960" == str(x)
+        assert time(23, 59, 59) == x
```

Comment:
I'm not sure what to do here. Should I keep the original value in `original_string` or update it?

Issue arises because DICOM Standard allows seconds in the range 0 to 60, but datetime only allows 0 to 59

Reply:
Also unsure. According to the standard, this is only valid if it is a lep second - and that happens no more than once a year. So it _could_ be valid (given it is related to the correct day), but that's quite unlikely. In most cases it won't matter anyway, so I'm not sure if we even should warn here at all (or if it is even a real-life use cases). All things considered, I would probably keep the `original_string`, so it can be re-encoded as is.

Reply:
> According to the standard, this is only valid if it is a leap second

Interesting. Date/time stuff is so messy.

I'll add another 'unsure' but also think keeping `original_string` is always a good idea if possible.